src/Api/Auth/Models/Request/TwoFactorRequestModels.cs

```diff
@@ -225,6 +225,14 @@ public User ToUser(User extistingUser)
         extistingUser.SetTwoFactorProviders(providers);
         return extistingUser;
     }
+
+    public override IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+    {
+        if (string.IsNullOrEmpty(Secret) && string.IsNullOrEmpty(AuthRequestAccessCode) && string.IsNullOrEmpty((SsoEmail2FaSessionToken)))
+        {
+            yield return new ValidationResult("MasterPasswordHash, OTP, AccessCode, or SsoEmail2faSessionToken must be supplied.");
+        }
+    }
```

Comment:
Should this call into the base `Validate` method? I'm not sure if it actually does anything or not.